src/addons/proxymonitor/metaservice/src/main/java/ee/ria/xroad/proxy/serverproxy/ProxyMonitorServiceHandlerImpl.java

```diff
@@ -123,13 +135,48 @@ public void startHandling(HttpServletRequest servletRequest,
         root.getMetrics().add(version);
 
         if (client != null) {
-            root.getMetrics().add(client.getMetrics());
+            root.getMetrics().add(client.getMetrics(getMetricNames(proxyRequestMessage)));
         }
 
         SoapMessageImpl result = createResponse(requestMessage.getSoap(), metricsResponse);
         responseEncoder.soap(result, Collections.emptyMap());
     }
 
+    /**
+     * Read requested monitoring parameter names from SOAP body. Returns empty list if no explicit metric names defined.
+     *
+     * @param proxyRequestMessage
+     * @return
+     */
+    private List<String> getMetricNames(ProxyMessage proxyRequestMessage) throws Exception {
+        List<String> metricNames = new ArrayList<>();
+
+        Document doc = parse(proxyRequestMessage);
+        NodeList nl = doc.getElementsByTagNameNS(NS_MONITORING, MONITOR_REQ_PARAM_NODE_NAME);
+
+        for (int i = 0; i < nl.getLength(); i++) {
+            Node n = nl.item(i);
+            String val = n.getFirstChild().getNodeValue();
+            metricNames.add(val);
```

Comment:
Could this be written on one line without temp variables e.g. 
```
metricNames.add(nl.item(i).getFirstChild().getNodeValue());
```

Reply:
Yep, was in longer form for easier debugging. Corrected.